### PR TITLE
HTBHF-689 Added hmrc entities and repository.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Adult.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    @Column(name = "award_date")
+    private LocalDate awardDate;
+
+    @Size(max = 35)
+    @Column(name = "surname")
+    private String surname;
+
+    @Size(max = 35)
+    @Column(name = "first_forename")
+    private String firstForeName;
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Adult.java]: Forename in this field and the one below should not have a capital N for name.

File: api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Child.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+public class Child extends BaseEntity {
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "hmrc_ctc_household_id", nullable = false)
+    private Household household;
+
+    @Size(max = 35)
+    @Column(name = "first_forename")
+    private String firstForeName;
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Child.java]: Same here, lower case n.

Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Child.java]: Updated.

File: api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Adult.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/hmrc/entity/Adult.java]: Updated.

**Review by mikeSmith08 — APPROVED** [review]

**Review by gethyn1 — APPROVED** [review]

**Pull request merged**